octobot_trading/exchanges/connectors/ccxt_exchange.py

```diff
@@ -301,91 +493,474 @@ async def get_all_currencies_price_ticker(self, **kwargs: dict) -> typing.Option
+                    trades.append(order)
+            return trades, None
+        except Exception as e:
+            return None, f"Failed to fetch recent trades using get_my_recent_trades_using_closed_orders: {e}"
+
+    def cut_recent_trades_pagination_limit(self, limit: int) -> typing.Optional[int]:
+        if self.connector_config.MAX_RECENT_TRADES_PAGINATION_LIMIT:
+            return min(self.connector_config.MAX_RECENT_TRADES_PAGINATION_LIMIT, limit)
         else:
-            raise octobot_trading.errors.NotSupported("This exchange doesn't support fetchMyTrades nor fetchTrades")
+            return limit
 
     async def cancel_order(self, order_id: str, symbol: str = None, **kwargs: dict) -> bool:
-        cancel_resp = None
+        defined_methods = self.connector_config.CANCEL_ORDERS_METHODS
+        messages = ""
+        if enums.CCXTExchangeConfigMethods.CANCEL_ORDER_DEFAULT.value in defined_methods:
+            success, error_message = await self.cancel_order_default(order_id, symbol=symbol, **kwargs)
+            if success:                    
+                return True
+            else:
+                messages = error_message or messages
+        if enums.CCXTExchangeConfigMethods.CANCEL_ORDER_DEFAULT.value in defined_methods:
+            success, await self.cancel_stop_order_using_stop_loss_endpoint(order_id, symbol=symbol, **kwargs)
+            if success:                    
+                return True
+            else:
+                messages = error_message + messages if error_message else messages
+        if messages != "":
+            raise octobot_trading.errors.NotSupported(messages)
+        self.logger.warning(f"Failed to cancel order {symbol} {order_id} - order was not found")
+        return False # order not found
+
+    async def cancel_order_default(self, order_id: str, symbol: str = None, **kwargs: dict) -> bool:
+        success = False
+        cancel_resp_message = None
         try:
+            cancel_resp = None
             with self.error_describer():
                 cancel_resp = await self.client.cancel_order(order_id, symbol=symbol, params=kwargs)
-            try:
-                cancelled_order = await self.exchange_manager.exchange.get_order(order_id, symbol=symbol)
-                return cancelled_order is None or personal_data.parse_is_cancelled(cancelled_order)
-            except ccxt.OrderNotFound:
-                # Order is not found: it has successfully been cancelled (some exchanges don't allow to
-                # get a cancelled order).
-                return True
+            cancel_resp_message = f" - Cancel response: {cancel_resp or 'no response'}"
+            is_canceled, _ =  await self.check_if_canceled(order_id, symbol)
+            if is_canceled:
+                return
+            await asyncio.sleep(20) # try again - some exchanges need a while to update the order (bybit for example)
```

Comment:
Why should we wait `20` seconds?

Reply:
for some reason bybit takes 15 -20 seconds to respond the order with status canceled - before that status is still open